
> securekey@0.1.0 test
> react-scripts test --watchAll=false --silent --coverage --verbose --silent=false

  console.error
    Error validating user context:  Error: Validation Error
        at Object.<anonymous> (/Users/shaillyjain/Projects/securekey/src/utils/firestoredb1.test.js:96:31)
        at Promise.then.completed (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/utils.js:391:28)
        at new Promise (<anonymous>)
        at callAsyncCircusFn (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/utils.js:316:10)
        at _callCircusTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:218:40)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
        at _runTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:155:3)
        at _runTestsForDescribeBlock (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:66:9)
        at _runTestsForDescribeBlock (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:60:9)
        at _runTestsForDescribeBlock (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:60:9)
        at run (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:25:3)
        at runAndTransformResultsToJestFormat (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)
        at jestAdapter (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:82:19)
        at runTestInternal (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/runTest.js:389:16)
        at runTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/runTest.js:475:34)
        at Object.worker (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/testWorker.js:133:12)

      51 |       })
      52 |       .catch((error) => {
    > 53 |         console.error("Error validating user context: ", error);
         |                 ^
      54 |         reject({ message: "Error validating user context" });
      55 |       });
      56 |   });

      at src/utils/firestoredb.js:53:17

  console.error
    Error validating user context:  User context not found

      51 |       })
      52 |       .catch((error) => {
    > 53 |         console.error("Error validating user context: ", error);
         |                 ^
      54 |         reject({ message: "Error validating user context" });
      55 |       });
      56 |   });

      at src/utils/firestoredb.js:53:17

  console.error
    Error validating user context:  User not found in user context

      51 |       })
      52 |       .catch((error) => {
    > 53 |         console.error("Error validating user context: ", error);
         |                 ^
      54 |         reject({ message: "Error validating user context" });
      55 |       });
      56 |   });

      at src/utils/firestoredb.js:53:17

  console.error
    Error validating user context:  User not found in user context

      51 |       })
      52 |       .catch((error) => {
    > 53 |         console.error("Error validating user context: ", error);
         |                 ^
      54 |         reject({ message: "Error validating user context" });
      55 |       });
      56 |   });

      at src/utils/firestoredb.js:53:17

  console.error
    Error getting documents from Firestore:  Error: Firestore Error
        at Object.<anonymous> (/Users/shaillyjain/Projects/securekey/src/utils/firestoredb1.test.js:111:31)
        at Promise.then.completed (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/utils.js:391:28)
        at new Promise (<anonymous>)
        at callAsyncCircusFn (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/utils.js:316:10)
        at _callCircusTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:218:40)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
        at _runTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:155:3)
        at _runTestsForDescribeBlock (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:66:9)
        at _runTestsForDescribeBlock (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:60:9)
        at _runTestsForDescribeBlock (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:60:9)
        at run (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:25:3)
        at runAndTransformResultsToJestFormat (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)
        at jestAdapter (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:82:19)
        at runTestInternal (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/runTest.js:389:16)
        at runTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/runTest.js:475:34)
        at Object.worker (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/testWorker.js:133:12)

      46 |           })
      47 |           .catch((error) => {
    > 48 |             console.error("Error getting documents from Firestore: ", error);
         |                     ^
      49 |             reject({ message: "Error getting documents from Firestore" });
      50 |           });
      51 |       })

      at src/utils/firestoredb.js:48:21

  console.error
    Error validating user context:  Username not found in user context

      51 |       })
      52 |       .catch((error) => {
    > 53 |         console.error("Error validating user context: ", error);
         |                 ^
      54 |         reject({ message: "Error validating user context" });
      55 |       });
      56 |   });

      at src/utils/firestoredb.js:53:17

  console.error
    Error validating user context:  Username not found in user context

      51 |       })
      52 |       .catch((error) => {
    > 53 |         console.error("Error validating user context: ", error);
         |                 ^
      54 |         reject({ message: "Error validating user context" });
      55 |       });
      56 |   });

      at src/utils/firestoredb.js:53:17

  console.error
    Error validating user context:  Password not found in user context

      51 |       })
      52 |       .catch((error) => {
    > 53 |         console.error("Error validating user context: ", error);
         |                 ^
      54 |         reject({ message: "Error validating user context" });
      55 |       });
      56 |   });

      at src/utils/firestoredb.js:53:17

  console.error
    Error validating user context:  Password not found in user context

      51 |       })
      52 |       .catch((error) => {
    > 53 |         console.error("Error validating user context: ", error);
         |                 ^
      54 |         reject({ message: "Error validating user context" });
      55 |       });
      56 |   });

      at src/utils/firestoredb.js:53:17

  console.error
    Error validating user context:  Error: Validation Error
        at Object.<anonymous> (/Users/shaillyjain/Projects/securekey/src/utils/firestoredb1.test.js:164:31)
        at Promise.then.completed (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/utils.js:391:28)
        at new Promise (<anonymous>)
        at callAsyncCircusFn (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/utils.js:316:10)
        at _callCircusTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:218:40)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
        at _runTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:155:3)
        at _runTestsForDescribeBlock (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:66:9)
        at _runTestsForDescribeBlock (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:60:9)
        at _runTestsForDescribeBlock (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:60:9)
        at run (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:25:3)
        at runAndTransformResultsToJestFormat (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)
        at jestAdapter (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:82:19)
        at runTestInternal (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/runTest.js:389:16)
        at runTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/runTest.js:475:34)
        at Object.worker (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/testWorker.js:133:12)

      93 |       })
      94 |       .catch((error) => {
    > 95 |         console.error("Error validating user context: ", error);
         |                 ^
      96 |         reject({ message: "Error validating user context" });
      97 |       });
      98 |   });

      at src/utils/firestoredb.js:95:17

  console.error
    Error getting documents from Firestore:  Error: Firestore Error
        at Object.<anonymous> (/Users/shaillyjain/Projects/securekey/src/utils/firestoredb1.test.js:183:31)
        at Promise.then.completed (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/utils.js:391:28)
        at new Promise (<anonymous>)
        at callAsyncCircusFn (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/utils.js:316:10)
        at _callCircusTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:218:40)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
        at _runTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:155:3)
        at _runTestsForDescribeBlock (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:66:9)
        at _runTestsForDescribeBlock (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:60:9)
        at _runTestsForDescribeBlock (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:60:9)
        at run (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:25:3)
        at runAndTransformResultsToJestFormat (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)
        at jestAdapter (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:82:19)
        at runTestInternal (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/runTest.js:389:16)
        at runTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/runTest.js:475:34)
        at Object.worker (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/testWorker.js:133:12)

      88 |           })
      89 |           .catch((error) => {
    > 90 |             console.error("Error getting documents from Firestore: ", error);
         |                     ^
      91 |             reject({ message: "Error getting documents from Firestore" });
      92 |           });
      93 |       })

      at src/utils/firestoredb.js:90:21

  console.error
    Error getting documents from Firestore:  error

      46 |           })
      47 |           .catch((error) => {
    > 48 |             console.error("Error getting documents from Firestore: ", error);
         |                     ^
      49 |             reject({ message: "Error getting documents from Firestore" });
      50 |           });
      51 |       })

      at src/utils/firestoredb.js:48:21

  console.error
    Error validating user context:  User context not found

      93 |       })
      94 |       .catch((error) => {
    > 95 |         console.error("Error validating user context: ", error);
         |                 ^
      96 |         reject({ message: "Error validating user context" });
      97 |       });
      98 |   });

      at src/utils/firestoredb.js:95:17

  console.error
    Error validating user context:  User not found in user context

      93 |       })
      94 |       .catch((error) => {
    > 95 |         console.error("Error validating user context: ", error);
         |                 ^
      96 |         reject({ message: "Error validating user context" });
      97 |       });
      98 |   });

      at src/utils/firestoredb.js:95:17

  console.error
    Error validating user context:  User not found in user context

      93 |       })
      94 |       .catch((error) => {
    > 95 |         console.error("Error validating user context: ", error);
         |                 ^
      96 |         reject({ message: "Error validating user context" });
      97 |       });
      98 |   });

      at src/utils/firestoredb.js:95:17

  console.error
    Error validating user context:  Username not found in user context

      93 |       })
      94 |       .catch((error) => {
    > 95 |         console.error("Error validating user context: ", error);
         |                 ^
      96 |         reject({ message: "Error validating user context" });
      97 |       });
      98 |   });

      at src/utils/firestoredb.js:95:17

  console.error
    Error validating user context:  Username not found in user context

      93 |       })
      94 |       .catch((error) => {
    > 95 |         console.error("Error validating user context: ", error);
         |                 ^
      96 |         reject({ message: "Error validating user context" });
      97 |       });
      98 |   });

      at src/utils/firestoredb.js:95:17

  console.error
    Error validating user context:  Password not found in user context

      93 |       })
      94 |       .catch((error) => {
    > 95 |         console.error("Error validating user context: ", error);
         |                 ^
      96 |         reject({ message: "Error validating user context" });
      97 |       });
      98 |   });

      at src/utils/firestoredb.js:95:17

  console.error
    Error validating user context:  Password not found in user context

      93 |       })
      94 |       .catch((error) => {
    > 95 |         console.error("Error validating user context: ", error);
         |                 ^
      96 |         reject({ message: "Error validating user context" });
      97 |       });
      98 |   });

      at src/utils/firestoredb.js:95:17

  console.error
    Error decrypting passkey:  Error: Decryption Error
        at /Users/shaillyjain/Projects/securekey/src/utils/firestoredb1.test.js:208:54
        at /Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:449:39
        at /Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:457:13
        at mockConstructor (/Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:170:19)
        at /Users/shaillyjain/Projects/securekey/src/utils/firestoredb.js:82:37
        at processTicksAndRejections (node:internal/process/task_queues:95:5)

      83 |               resolve(password);
      84 |             } catch (error) {
    > 85 |               console.error("Error decrypting passkey: ", error);
         |                       ^
      86 |               reject({ message: "Error decrypting passkey" });
      87 |             }
      88 |           })

      at src/utils/firestoredb.js:85:23

  console.error
    Error decrypting passkey:  Error: decryption error
        at /Users/shaillyjain/Projects/securekey/src/utils/firestoredb.test.js:353:13
        at /Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:449:39
        at /Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:457:13
        at mockConstructor (/Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:170:19)
        at decrypt (eval at _createMockFunction (/Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:630:31), <anonymous>:3:51)
        at /Users/shaillyjain/Projects/securekey/src/utils/firestoredb.js:82:37
        at processTicksAndRejections (node:internal/process/task_queues:95:5)

      83 |               resolve(password);
      84 |             } catch (error) {
    > 85 |               console.error("Error decrypting passkey: ", error);
         |                       ^
      86 |               reject({ message: "Error decrypting passkey" });
      87 |             }
      88 |           })

      at src/utils/firestoredb.js:85:23

  console.error
    Error getting documents from Firestore:  error

      88 |           })
      89 |           .catch((error) => {
    > 90 |             console.error("Error getting documents from Firestore: ", error);
         |                     ^
      91 |             reject({ message: "Error getting documents from Firestore" });
      92 |           });
      93 |       })

      at src/utils/firestoredb.js:90:21

  console.error
    Error validating user context:  User context not found

      155 |       })
      156 |       .catch((error) => {
    > 157 |         console.error("Error validating user context: ", error);
          |                 ^
      158 |         reject({ message: "Error validating user context" });
      159 |       });
      160 |   });

      at src/utils/firestoredb.js:157:17

  console.error
    Error validating user context:  User not found in user context

      155 |       })
      156 |       .catch((error) => {
    > 157 |         console.error("Error validating user context: ", error);
          |                 ^
      158 |         reject({ message: "Error validating user context" });
      159 |       });
      160 |   });

      at src/utils/firestoredb.js:157:17

  console.error
    Error validating user context:  User not found in user context

      155 |       })
      156 |       .catch((error) => {
    > 157 |         console.error("Error validating user context: ", error);
          |                 ^
      158 |         reject({ message: "Error validating user context" });
      159 |       });
      160 |   });

      at src/utils/firestoredb.js:157:17

  console.error
    Error validating user context:  Username not found in user context

      155 |       })
      156 |       .catch((error) => {
    > 157 |         console.error("Error validating user context: ", error);
          |                 ^
      158 |         reject({ message: "Error validating user context" });
      159 |       });
      160 |   });

      at src/utils/firestoredb.js:157:17

  console.error
    Error validating user context:  Username not found in user context

      155 |       })
      156 |       .catch((error) => {
    > 157 |         console.error("Error validating user context: ", error);
          |                 ^
      158 |         reject({ message: "Error validating user context" });
      159 |       });
      160 |   });

      at src/utils/firestoredb.js:157:17

  console.error
    Error validating user context:  Password not found in user context

      155 |       })
      156 |       .catch((error) => {
    > 157 |         console.error("Error validating user context: ", error);
          |                 ^
      158 |         reject({ message: "Error validating user context" });
      159 |       });
      160 |   });

      at src/utils/firestoredb.js:157:17

  console.error
    Error validating user context:  Password not found in user context

      155 |       })
      156 |       .catch((error) => {
    > 157 |         console.error("Error validating user context: ", error);
          |                 ^
      158 |         reject({ message: "Error validating user context" });
      159 |       });
      160 |   });

      at src/utils/firestoredb.js:157:17

  console.error
    Warning: `ReactDOMTestUtils.act` is deprecated in favor of `React.act`. Import `act` from `react` instead of `react-dom/test-utils`. See https://react.dev/warnings/react-dom-test-utils for more info.

      12 |
      13 | const renderNavbarWithUserContext = (userContextValue) => {
    > 14 |   return render(
         |                ^
      15 |     <UserContext.Provider value={userContextValue}>
      16 |       <Navbar />
      17 |     </UserContext.Provider>

      at printWarning (node_modules/react-dom/cjs/react-dom-test-utils.development.js:71:30)
      at error (node_modules/react-dom/cjs/react-dom-test-utils.development.js:45:7)
      at actWithWarning (node_modules/react-dom/cjs/react-dom-test-utils.development.js:1736:7)
      at node_modules/@testing-library/react/dist/act-compat.js:63:25
      at renderRoot (node_modules/@testing-library/react/dist/pure.js:159:26)
      at render (node_modules/@testing-library/react/dist/pure.js:246:10)
      at renderNavbarWithUserContext (src/components/Navbar.test.js:14:16)
      at Object.<anonymous> (src/components/Navbar.test.js:29:5)

  console.error
    Error saving passkey to Firestore:  error

      144 |                 })
      145 |                 .catch((error) => {
    > 146 |                   console.error("Error saving passkey to Firestore: ", error);
          |                           ^
      147 |                   reject({ message: "Error saving passkey to Firestore" });
      148 |                 });
      149 |             }

      at src/utils/firestoredb.js:146:27

  console.error
    Error getting documents from Firestore:  error

      150 |           })
      151 |           .catch((error) => {
    > 152 |             console.error("Error getting documents from Firestore: ", error);
          |                     ^
      153 |             reject({ message: "Error getting documents from Firestore" });
      154 |           });
      155 |       })

      at src/utils/firestoredb.js:152:21

  console.error
    Error encrypting passkey:  Error: encryption error
        at /Users/shaillyjain/Projects/securekey/src/utils/firestoredb.test.js:675:13
        at /Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:449:39
        at /Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:457:13
        at mockConstructor (/Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:170:19)
        at encrypt (eval at _createMockFunction (/Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:630:31), <anonymous>:3:51)
        at /Users/shaillyjain/Projects/securekey/src/utils/firestoredb.js:112:31
        at processTicksAndRejections (node:internal/process/task_queues:95:5)

      112 |           ciphertext = encrypt({ plaintext: password, key: secretKey });
      113 |         } catch (error) {
    > 114 |           console.error("Error encrypting passkey: ", error);
          |                   ^
      115 |           reject({ message: "Error encrypting passkey" });
      116 |           return;
      117 |         }

      at src/utils/firestoredb.js:114:19

  console.error
    Error validating user context:  User context not found

      221 |       })
      222 |       .catch((error) => {
    > 223 |         console.error("Error validating user context: ", error);
          |                 ^
      224 |         reject({ message: "Error validating user context" });
      225 |       });
      226 |   });

      at src/utils/firestoredb.js:223:17

  console.error
    Error validating user context:  User not found in user context

      221 |       })
      222 |       .catch((error) => {
    > 223 |         console.error("Error validating user context: ", error);
          |                 ^
      224 |         reject({ message: "Error validating user context" });
      225 |       });
      226 |   });

      at src/utils/firestoredb.js:223:17

  console.error
    Error validating user context:  User not found in user context

      221 |       })
      222 |       .catch((error) => {
    > 223 |         console.error("Error validating user context: ", error);
          |                 ^
      224 |         reject({ message: "Error validating user context" });
      225 |       });
      226 |   });

      at src/utils/firestoredb.js:223:17

  console.error
    Error validating user context:  Username not found in user context

      221 |       })
      222 |       .catch((error) => {
    > 223 |         console.error("Error validating user context: ", error);
          |                 ^
      224 |         reject({ message: "Error validating user context" });
      225 |       });
      226 |   });

      at src/utils/firestoredb.js:223:17

  console.error
    Error validating user context:  Username not found in user context

      221 |       })
      222 |       .catch((error) => {
    > 223 |         console.error("Error validating user context: ", error);
          |                 ^
      224 |         reject({ message: "Error validating user context" });
      225 |       });
      226 |   });

      at src/utils/firestoredb.js:223:17

  console.error
    Error validating user context:  Password not found in user context

      221 |       })
      222 |       .catch((error) => {
    > 223 |         console.error("Error validating user context: ", error);
          |                 ^
      224 |         reject({ message: "Error validating user context" });
      225 |       });
      226 |   });

      at src/utils/firestoredb.js:223:17

  console.error
    Error validating user context:  Password not found in user context

      221 |       })
      222 |       .catch((error) => {
    > 223 |         console.error("Error validating user context: ", error);
          |                 ^
      224 |         reject({ message: "Error validating user context" });
      225 |       });
      226 |   });

      at src/utils/firestoredb.js:223:17

  console.error
    Warning: `ReactDOMTestUtils.act` is deprecated in favor of `React.act`. Import `act` from `react` instead of `react-dom/test-utils`. See https://react.dev/warnings/react-dom-test-utils for more info.

      18 |
      19 | const renderKeysListWithUserContext = (userContextValue, props = {}) => {
    > 20 |   return render(
         |                ^
      21 |     <UserContext.Provider value={userContextValue}>
      22 |       <KeysList {...props} />
      23 |     </UserContext.Provider>

      at printWarning (node_modules/react-dom/cjs/react-dom-test-utils.development.js:71:30)
      at error (node_modules/react-dom/cjs/react-dom-test-utils.development.js:45:7)
      at actWithWarning (node_modules/react-dom/cjs/react-dom-test-utils.development.js:1736:7)
      at node_modules/@testing-library/react/dist/act-compat.js:63:25
      at renderRoot (node_modules/@testing-library/react/dist/pure.js:159:26)
      at render (node_modules/@testing-library/react/dist/pure.js:246:10)
      at renderKeysListWithUserContext (src/components/KeyList.test.js:20:16)
      at Object.<anonymous> (src/components/KeyList.test.js:43:5)

  console.error
    Error updating passkey in Firestore:  error

      211 |               })
      212 |               .catch((error) => {
    > 213 |                 console.error("Error updating passkey in Firestore: ", error);
          |                         ^
      214 |                 reject({ message: "Error updating passkey in Firestore" });
      215 |               });
      216 |           })

      at src/utils/firestoredb.js:213:25

  console.error
    Error getting documents from Firestore:  error

      216 |           })
      217 |           .catch((error) => {
    > 218 |             console.error("Error getting documents from Firestore: ", error);
          |                     ^
      219 |             reject({ message: "Error getting documents from Firestore" });
      220 |           });
      221 |       })

      at src/utils/firestoredb.js:218:21

  console.error
    Error encrypting passkey:  Error: encryption error
        at /Users/shaillyjain/Projects/securekey/src/utils/firestoredb.test.js:993:13
        at /Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:449:39
        at /Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:457:13
        at mockConstructor (/Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:170:19)
        at encrypt (eval at _createMockFunction (/Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:630:31), <anonymous>:3:51)
        at /Users/shaillyjain/Projects/securekey/src/utils/firestoredb.js:174:31
        at processTicksAndRejections (node:internal/process/task_queues:95:5)

      174 |           ciphertext = encrypt({ plaintext: password, key: secretKey });
      175 |         } catch (error) {
    > 176 |           console.error("Error encrypting passkey: ", error);
          |                   ^
      177 |           reject({ message: "Error encrypting passkey" });
      178 |           return;
      179 |         }

      at src/utils/firestoredb.js:176:19

  console.error
    Error validating user context:  User context not found

      261 |       })
      262 |       .catch((error) => {
    > 263 |         console.error("Error validating user context: ", error);
          |                 ^
      264 |         reject({ message: "Error validating user context" });
      265 |       });
      266 |   });

      at src/utils/firestoredb.js:263:17

  console.error
    Error validating user context:  User not found in user context

      261 |       })
      262 |       .catch((error) => {
    > 263 |         console.error("Error validating user context: ", error);
          |                 ^
      264 |         reject({ message: "Error validating user context" });
      265 |       });
      266 |   });

      at src/utils/firestoredb.js:263:17

  console.error
    Error validating user context:  User not found in user context

      261 |       })
      262 |       .catch((error) => {
    > 263 |         console.error("Error validating user context: ", error);
          |                 ^
      264 |         reject({ message: "Error validating user context" });
      265 |       });
      266 |   });

      at src/utils/firestoredb.js:263:17

  console.error
    Error validating user context:  Username not found in user context

      261 |       })
      262 |       .catch((error) => {
    > 263 |         console.error("Error validating user context: ", error);
          |                 ^
      264 |         reject({ message: "Error validating user context" });
      265 |       });
      266 |   });

      at src/utils/firestoredb.js:263:17

  console.error
    Error validating user context:  Username not found in user context

      261 |       })
      262 |       .catch((error) => {
    > 263 |         console.error("Error validating user context: ", error);
          |                 ^
      264 |         reject({ message: "Error validating user context" });
      265 |       });
      266 |   });

      at src/utils/firestoredb.js:263:17

  console.error
    Error validating user context:  Password not found in user context

      261 |       })
      262 |       .catch((error) => {
    > 263 |         console.error("Error validating user context: ", error);
          |                 ^
      264 |         reject({ message: "Error validating user context" });
      265 |       });
      266 |   });

      at src/utils/firestoredb.js:263:17

  console.error
    Error validating user context:  Password not found in user context

      261 |       })
      262 |       .catch((error) => {
    > 263 |         console.error("Error validating user context: ", error);
          |                 ^
      264 |         reject({ message: "Error validating user context" });
      265 |       });
      266 |   });

      at src/utils/firestoredb.js:263:17

  console.error
    Error deleting passkey from Firestore:  error

      251 |               })
      252 |               .catch((error) => {
    > 253 |                 console.error("Error deleting passkey from Firestore: ", error);
          |                         ^
      254 |                 reject({ message: "Error deleting passkey from Firestore" });
      255 |               });
      256 |           })

      at src/utils/firestoredb.js:253:25

  console.error
    Error getting documents from Firestore:  error

      256 |           })
      257 |           .catch((error) => {
    > 258 |             console.error("Error getting documents from Firestore: ", error);
          |                     ^
      259 |             reject({ message: "Error getting documents from Firestore" });
      260 |           });
      261 |       })

      at src/utils/firestoredb.js:258:21

  console.error
    Error validating user context:  User context not found

      317 |       })
      318 |       .catch((error) => {
    > 319 |         console.error("Error validating user context: ", error);
          |                 ^
      320 |         reject({ message: "Error validating user context" });
      321 |       });
      322 |   });

      at src/utils/firestoredb.js:319:17

  console.error
    Error validating user context:  User not found in user context

      317 |       })
      318 |       .catch((error) => {
    > 319 |         console.error("Error validating user context: ", error);
          |                 ^
      320 |         reject({ message: "Error validating user context" });
      321 |       });
      322 |   });

      at src/utils/firestoredb.js:319:17

  console.error
    Error validating user context:  User not found in user context

      317 |       })
      318 |       .catch((error) => {
    > 319 |         console.error("Error validating user context: ", error);
          |                 ^
      320 |         reject({ message: "Error validating user context" });
      321 |       });
      322 |   });

      at src/utils/firestoredb.js:319:17

  console.error
    Error validating user context:  Username not found in user context

      317 |       })
      318 |       .catch((error) => {
    > 319 |         console.error("Error validating user context: ", error);
          |                 ^
      320 |         reject({ message: "Error validating user context" });
      321 |       });
      322 |   });

      at src/utils/firestoredb.js:319:17

  console.error
    Error validating user context:  Username not found in user context

      317 |       })
      318 |       .catch((error) => {
    > 319 |         console.error("Error validating user context: ", error);
          |                 ^
      320 |         reject({ message: "Error validating user context" });
      321 |       });
      322 |   });

      at src/utils/firestoredb.js:319:17

  console.error
    Error validating user context:  Password not found in user context

      317 |       })
      318 |       .catch((error) => {
    > 319 |         console.error("Error validating user context: ", error);
          |                 ^
      320 |         reject({ message: "Error validating user context" });
      321 |       });
      322 |   });

      at src/utils/firestoredb.js:319:17

  console.error
    Error validating user context:  Password not found in user context

      317 |       })
      318 |       .catch((error) => {
    > 319 |         console.error("Error validating user context: ", error);
          |                 ^
      320 |         reject({ message: "Error validating user context" });
      321 |       });
      322 |   });

      at src/utils/firestoredb.js:319:17

  console.error
    Warning: An update to KeysList inside a test was not wrapped in act(...).
    
    When testing, code that causes React state updates should be wrapped into act(...):
    
    act(() => {
      /* fire events that update state */
    });
    /* assert on the output */
    
    This ensures that you're testing the behavior the user would see in the browser. Learn more at https://reactjs.org/link/wrap-tests-with-act
        at KeysList (/Users/shaillyjain/Projects/securekey/src/components/KeysList.js:13:33)

      62 |       })
      63 |       .finally(() => {
    > 64 |         setLoading(false);
         |         ^
      65 |       });
      66 |   }, []);
      67 |

      at printWarning (node_modules/react-dom/cjs/react-dom.development.js:86:30)
      at error (node_modules/react-dom/cjs/react-dom.development.js:60:7)
      at warnIfUpdatesNotWrappedWithActDEV (node_modules/react-dom/cjs/react-dom.development.js:27628:9)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:25547:5)
      at setLoading (node_modules/react-dom/cjs/react-dom.development.js:16708:7)
      at src/components/KeysList.js:64:9

  console.error
    Error decrypting passkey:  Error: decryption error
        at /Users/shaillyjain/Projects/securekey/src/utils/firestoredb.test.js:1379:13
        at /Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:449:39
        at /Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:457:13
        at mockConstructor (/Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:170:19)
        at decrypt (eval at _createMockFunction (/Users/shaillyjain/Projects/securekey/node_modules/jest-mock/build/index.js:630:31), <anonymous>:3:51)
        at /Users/shaillyjain/Projects/securekey/src/utils/firestoredb.js:295:39
        at processTicksAndRejections (node:internal/process/task_queues:95:5)

      303 |                 });
      304 |               } catch (error) {
    > 305 |                 console.error("Error decrypting passkey: ", error);
          |                         ^
      306 |                 return reject({ message: "Error decrypting passkey" });
      307 |               }
      308 |             }

      at src/utils/firestoredb.js:305:25

  console.error
    Error getting documents from Firestore:  error

      312 |           })
      313 |           .catch((error) => {
    > 314 |             console.error("Error getting documents from Firestore: ", error);
          |                     ^
      315 |             reject({ message: "Error getting documents from Firestore" });
      316 |           });
      317 |       })

      at src/utils/firestoredb.js:314:21

  console.error
    Error fetching keys Error: Server down
        at Object.<anonymous> (/Users/shaillyjain/Projects/securekey/src/components/KeyList.test.js:103:35)
        at Promise.then.completed (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/utils.js:391:28)
        at new Promise (<anonymous>)
        at callAsyncCircusFn (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/utils.js:316:10)
        at _callCircusTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:218:40)
        at _runTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:155:3)
        at _runTestsForDescribeBlock (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:66:9)
        at _runTestsForDescribeBlock (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:60:9)
        at run (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/run.js:25:3)
        at runAndTransformResultsToJestFormat (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)
        at jestAdapter (/Users/shaillyjain/Projects/securekey/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:82:19)
        at runTestInternal (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/runTest.js:389:16)
        at runTest (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/runTest.js:475:34)
        at Object.worker (/Users/shaillyjain/Projects/securekey/node_modules/jest-runner/build/testWorker.js:133:12)

      54 |       })
      55 |       .catch((error) => {
    > 56 |         console.error("Error fetching keys", error);
         |                 ^
      57 |         if (error.message) {
      58 |           setError(error.message);
      59 |         } else {

      at src/components/KeysList.js:56:17

  console.error
    Warning: An update to KeysList inside a test was not wrapped in act(...).
    
    When testing, code that causes React state updates should be wrapped into act(...):
    
    act(() => {
      /* fire events that update state */
    });
    /* assert on the output */
    
    This ensures that you're testing the behavior the user would see in the browser. Learn more at https://reactjs.org/link/wrap-tests-with-act
        at KeysList (/Users/shaillyjain/Projects/securekey/src/components/KeysList.js:13:33)

      56 |         console.error("Error fetching keys", error);
      57 |         if (error.message) {
    > 58 |           setError(error.message);
         |           ^
      59 |         } else {
      60 |           setError("Error fetching keys");
      61 |         }

      at printWarning (node_modules/react-dom/cjs/react-dom.development.js:86:30)
      at error (node_modules/react-dom/cjs/react-dom.development.js:60:7)
      at warnIfUpdatesNotWrappedWithActDEV (node_modules/react-dom/cjs/react-dom.development.js:27628:9)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:25547:5)
      at setError (node_modules/react-dom/cjs/react-dom.development.js:16708:7)
      at src/components/KeysList.js:58:11

----------------------|---------|----------|---------|---------|------------------------------------------------------------
File                  | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s                                          
----------------------|---------|----------|---------|---------|------------------------------------------------------------
All files             |   53.99 |    26.06 |   54.83 |   53.29 |                                                            
 src                  |       0 |        0 |       0 |       0 |                                                            
  App.js              |       0 |        0 |       0 |       0 | 10-14                                                      
  index.js            |       0 |      100 |     100 |       0 | 7-13                                                       
  reportWebVitals.js  |       0 |        0 |       0 |       0 | 1-8                                                        
 src/components       |   24.31 |    15.33 |   22.35 |   23.95 |                                                            
  AccountIcon.js      |      80 |       50 |     100 |      80 | 8                                                          
  AddPassKey.js       |       0 |        0 |       0 |       0 | 10-167                                                     
  AuthForm.js         |       0 |        0 |       0 |       0 | 11-173                                                     
  History.js          |       0 |        0 |       0 |       0 | 9-73                                                       
  KeyItem.js          |   27.41 |       25 |   17.64 |   26.22 | 32-64,70-87,107-238                                        
  KeysList.js         |   59.42 |    61.53 |      50 |   59.42 | 32-36,60,94-95,103,109-112,118-119,127-148,155,160,195-214 
  Loader.js           |     100 |       50 |     100 |     100 | 4                                                          
  Navbar.js           |     100 |      100 |     100 |     100 |                                                            
 src/context          |     100 |      100 |       0 |     100 |                                                            
  UserContext.js      |     100 |      100 |       0 |     100 |                                                            
 src/utils            |     100 |      100 |     100 |     100 |                                                            
  AccountIconsList.js |     100 |      100 |     100 |     100 |                                                            
  contextutil.js      |     100 |      100 |     100 |     100 |                                                            
  contextvalidator.js |     100 |      100 |     100 |     100 |                                                            
  cryptoutil.js       |     100 |      100 |     100 |     100 |                                                            
  dateutil.js         |     100 |      100 |     100 |     100 |                                                            
  debounce.js         |     100 |      100 |     100 |     100 |                                                            
  firebase.js         |     100 |      100 |     100 |     100 |                                                            
  firestoredb.js      |     100 |      100 |     100 |     100 |                                                            
  passwordutil.js     |     100 |      100 |     100 |     100 |                                                            
  str2color.js        |     100 |      100 |     100 |     100 |                                                            
----------------------|---------|----------|---------|---------|------------------------------------------------------------
